br/pkg/restore/split_client.go

```diff
@@ -299,82 +299,96 @@ func (c *pdClient) sendSplitRegionRequest(
 ) (*kvrpcpb.SplitRegionResponse, error) {
 	var splitErrors error
 	for i := 0; i < splitRegionMaxRetryTime; i++ {
-		var peer *metapb.Peer
-		// scanRegions may return empty Leader in https://github.com/tikv/pd/blob/v4.0.8/server/grpc_service.go#L524
-		// so wee also need check Leader.Id != 0
-		if regionInfo.Leader != nil && regionInfo.Leader.Id != 0 {
-			peer = regionInfo.Leader
-		} else {
-			if len(regionInfo.Region.Peers) == 0 {
-				return nil, multierr.Append(splitErrors,
-					errors.Annotatef(berrors.ErrRestoreNoPeer, "region[%d] doesn't have any peer", regionInfo.Region.GetId()))
-			}
-			peer = regionInfo.Region.Peers[0]
+		retry, result, err := sendSplitRegionRequest(c, ctx, regionInfo, keys, &splitErrors, i)
+		if retry {
+			continue
 		}
-		storeID := peer.GetStoreId()
-		store, err := c.GetStore(ctx, storeID)
 		if err != nil {
 			return nil, multierr.Append(splitErrors, err)
 		}
-		opt := grpc.WithInsecure()
-		if c.tlsConf != nil {
-			opt = grpc.WithTransportCredentials(credentials.NewTLS(c.tlsConf))
-		}
-		conn, err := grpc.Dial(store.GetAddress(), opt)
-		if err != nil {
-			return nil, multierr.Append(splitErrors, err)
+		if result != nil {
+			return result, nil
 		}
-		defer conn.Close()
-		client := tikvpb.NewTikvClient(conn)
-		resp, err := splitRegionWithFailpoint(ctx, regionInfo, peer, client, keys, c.isRawKv)
-		if err != nil {
-			return nil, multierr.Append(splitErrors, err)
+		return nil, errors.Trace(splitErrors)
```

Comment:
seems this line makes for loop not work

Reply:
this logic is the origin site in https://github.com/pingcap/tidb/blob/master/br/pkg/restore/split_client.go#L373

Reply:
okay